sdk/translation/ai-translation-document-rest/test/public/node/containerHelper.ts

```diff
@@ -0,0 +1,181 @@
+  containerName?: string | undefined,
+): Promise<string> {
+  if (containerName === undefined) {
+    containerName = recorder.variable("sourceContainer", `source-${getUniqueName()}`);
+  }
+  const containerClient = await createContainer(recorder, containerName, documents);
+
+  const sasUrl = await containerClient.generateSasUrl({
+    permissions: ContainerSASPermissions.parse("rwl"),
+    expiresOn: getDateOneDayAfter(),
+  });
+  return `${sasUrl}`;
+}
+
+export async function createTargetContainer(
+  recorder: Recorder,
+  containerName?: string | undefined,
+  documents?: TestDocument[],
+): Promise<string> {
+  if (containerName === undefined) {
+    containerName = recorder.variable("targetContainer", `target-${getUniqueName()}`);
+  }
+  const containerClient = await createContainer(recorder, containerName, documents);
+
+  const sasUrl = await containerClient.generateSasUrl({
+    permissions: ContainerSASPermissions.parse("rwl"),
+    expiresOn: getDateOneDayAfter(),
+  });
+  return `${sasUrl}`;
+}
+
+export async function createGlossaryContainer(recorder: Recorder): Promise<string> {
+  const glossaryName = "validGlossary.csv";
+  const glossaryContent = "test, glossaryTest";
+  const documents: TestDocument[] = [createTestDocument(glossaryName, glossaryContent)];
+  const containerName = recorder.variable("glossaryContainer", `glossary-${getUniqueName()}`);
+  const containerClient = await createContainer(recorder, containerName, documents);
+
+  const sasUrl = await containerClient.generateSasUrl({
+    permissions: ContainerSASPermissions.parse("rwl"),
+    expiresOn: getDateOneDayAfter(),
+  });
+
+  // Extract the base URL and query parameters
+  const urlParts = `${sasUrl}`.split("?");
+  const baseUrl = urlParts[0];
+  const queryParams = urlParts[1];
+
+  // Add the document name to the base URL
+  const newUrl = `${baseUrl}/${glossaryName}?${queryParams}`;
+  return `${newUrl}`;
+}
+
+export async function createTargetContainerWithInfo(
+  recorder: Recorder,
+  documents?: TestDocument[],
+): Promise<Map<string, string>> {
+  const containerName = recorder.variable("targetContainer", `target-${getUniqueName()}`);
+  const containerClient = await createContainer(recorder, containerName, documents);
+
+  const sasUrl = await containerClient.generateSasUrl({
+    permissions: ContainerSASPermissions.parse("rwl"),
+    expiresOn: getDateOneDayAfter(),
+  });
+
+  const sasUrlTest = recorder.variable("sasUrl", `${sasUrl}`);
+  const containerValuesMap: Map<string, string> = new Map();
+  containerValuesMap.set("sasUrl", sasUrlTest);
+  containerValuesMap.set("containerName", containerName);
+  return containerValuesMap;
+}
+
+async function createContainer(
+  recorder: Recorder,
+  containerName: string,
+  documents?: TestDocument[],
+): Promise<ContainerClient> {
+  const blobServiceClient: BlobServiceClient = BlobServiceClient.fromConnectionString(
+    env.DOCUMENT_TRANSLATION_CONNECTION_STRING as string,
```

Comment:
wondering if we can/should avoid using a connection string here (instead relying on user auth)? We are currently in the process of removing auth using connection strings from other packages as part of our Spring Grove response.

/cc @maorleger @minhanh-phan

Reply:
@timovv , can you please share an example of a project implementing it ?

Reply:
Since this is from storage blob service, I am curious if we should change the storage service first? or is there an alternative there?

Reply:
I don't have any specific example off the top of my head, but in theory you would use the storage URL instead of a connection string, and pass in the credential you are using for testing, e.g. `const blobServiceClient = new BlobServiceClient(env.DOCUMENT_TRANSLATION_STORAGE_ACCOUNT_URL, credential)`. You would need to update the `test-resources.json` to output the account URL instead of a connection string for this to work.

If it's too tricky, this isn't a blocker for your release IMHO; we can revisit later.

@qiaozha: not sure if I'm answering your question, but what we're trying to do as part of this security initiative is to remove the use of connection strings in our tests where possible. The Storage SDK itself won't change and will still provide APIs to use connection strings.

Reply:
Since this is a beta release, we will keep this as is for now and work in it before GA